presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotConfig.java

```diff
@@ -38,6 +38,7 @@
     public static final int DEFAULT_MIN_CONNECTIONS_PER_SERVER = 10;
     public static final int DEFAULT_THREAD_POOL_SIZE = 30;
     public static final int DEFAULT_NON_AGGREGATE_LIMIT_FOR_BROKER_QUERIES = 25_000;
+    public static final int DEFAULT_LIMIT_TO_PUSHDOWN_ORDER_BY_FOR_BROKER_QUERIES = 5_000;
```

Comment:
Would love to know more about where this 5K was inferred from ? Is there a Presto config we can refer to instead here ? 

Like do we know where and according to what config is Presto violating this order ? If it is a simple threshhold, perhaps we should refer to that threshhold here instead of creating our own, to avoid them being out of sync.

Reply:
This is from my local test, I observed the block to be 7kish.

Reply:
Basically, this seems a bit brittle. I am wondering if we can be more surer in when to trigger this. Like should this be disabled if the connector would return more than a page ? (Then perhaps we should fail the query if order-by-limit pushdown was done but more than one page is being returned because of a wide result set). 

Or is there a config ? 

I am basically wary of the case when the limit is lower than this number, we push down order-by-limit and still see the issue you faced originally or incorrect ordering. 

Disabling this outright would also be okay with me because that is also deterministic.

Reply:
I see, then in this case I can just change it to 1 to disable it.
I'm not sure if Presto has config to define the page size.